presto-jdbc/src/main/java/com/facebook/presto/jdbc/QueryExecutor.java

```diff
@@ -82,12 +82,17 @@ protected void finalize()
         close();
     }
 
-    static QueryExecutor create(String userAgent, JettyIoPool jettyIoPool)
+    static HttpClientConfig baseClientConfig()
+    {
+        return new HttpClientConfig()
+                .setConnectTimeout(new Duration(10, TimeUnit.SECONDS))
+                .setSocksProxy(getSystemSocksProxy());
+    }
+
+    static QueryExecutor create(String userAgent, HttpClientConfig clientConfig, JettyIoPool jettyIoPool)
```

Comment:
Could you please accept here `HttpClient` instead of `HttpClientConfig` and `JettyIOPool`. I would need to add some request filters to the HttpClient for the authentication based on the connection properties. So i will need to refactor this method anyway.